src/wasm-interpreter.h

```diff
@@ -2957,18 +2957,22 @@ class ModuleRunnerBase : public ExpressionRunner<SubType> {
   void initializeMemoryContents() {
     initializeMemorySizes();
 
-    Const zero;
-    zero.value = Literal(uint32_t(0));
-    zero.finalize();
-
     // apply active memory segments
     for (size_t i = 0, e = wasm.dataSegments.size(); i < e; ++i) {
       auto& segment = wasm.dataSegments[i];
       if (segment->isPassive) {
         continue;
       }
+
+      auto* memory = wasm.getMemory(segment->memory);
+
+      Const zero;
+      zero.value = Literal::makeFromInt32(0, memory->indexType);
```

Comment:
I guess we should probably update `indexType` -> `addressType` as a separate PR since that is what the spec calls it now.